### PR TITLE
Add capability to find APIKey filter secrets

### PR DESCRIPTION
**Pull request by knlambert**
## Description

We want to add the possibility to create APIKey filters in edge stack. Although, these keys have to be stored in a secret, so it's required to add their discovery into the emissary code base, along with the one available for OAuth2.

This PR implements the lookup for the APIKeys filters, and the registration of the associated secrets.

## Testing

Deployed in a test cluster & added units tests for this specific case.

## Checklist

 - [x] I made sure to update `CHANGELOG.md`.

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [x] This is unlikely to impact how Ambassador performs at scale.

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [x] My change is adequately tested.

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [x] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

 - [x] The changes in this PR have been reviewed for security concerns and adherence to security best practices.

### CODE REVIEW

File: cmd/entrypoint/secrets.go
Comment (LanceEa) [inline cmd/entrypoint/secrets.go:446]: unnecessary `return` since the function has no return type

File: docs/releaseNotes.yml
Comment (LanceEa) [inline docs/releaseNotes.yml:35]: In general, I'm thinking that we do not call this out here because it has no relevance to Emissary-ingress users.

@AliceProxy - thoughts on this?

**Review by LanceEa — CHANGES_REQUESTED** [review]

**Comment by LanceEa** [comment]
@knlambert - overall it looks good but we just need to nail down the design so let's finish that discussion in EdgeStack then the feedback above can be addressed.

**Comment by knlambert** [comment]
@AliceProxy ?

**Review by LanceEa — APPROVED** [review]
lgtm - let's get a second review and then it should be good.

File: CHANGELOG.md
Comment (AliceProxy) [inline CHANGELOG.md:86]: Typo but you can address this in a follow-up PR. Lets get this landed.

File: docs/releaseNotes.yml
Comment (AliceProxy) [reply docs/releaseNotes.yml:35]: agreed

File: docs/releaseNotes.yml
Comment (AliceProxy) [reply docs/releaseNotes.yml:44]: I don't think the release notes are required. That said if we do have edgestack relnotes then putting thme under `edgeStackNotes:` is appropriate although not really done anymore since I don't think the docs render this section anymore.

**Review by AliceProxy — APPROVED** [review]
LGTM with a follow-up PR for the typo and release notes removal.

**Review by LanceEa — APPROVED** [review]

**Pull request merged**
